.github/workflows/ci.yml

```
@@ -106,3 +106,126 @@ jobs:
+  publish-abis:
+    name: "Publish Contract ABIs"
+    runs-on: ubuntu-latest
+
+    if: github.event_name == 'push' && github.ref == 'refs/heads/main'
+    needs: [ci, slither]
+    permissions:
+      contents: write
+
+    steps:
+      - name: "Check out the repo"
+        uses: actions/checkout@v3
```

actionlint (1.7.9)

the runner of "actions/checkout@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)